src/redux/features/AssignrolesSlice.ts

```diff
@@ -0,0 +1,64 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+
+import { createAsyncThunk, createSlice, PayloadAction } from '@reduxjs/toolkit';
+import { DynamicData } from '../../@types/DynamicData';
+import API from '../../utils/api';
+interface rolesState {
+	data: DynamicData[];
+	isLoading: boolean;
+	error: string | null;
+}
+const initialState: rolesState = {
+	data: [],
+	isLoading: false,
+	error: null,
+};
+export const assignRoles = createAsyncThunk(
+	'assignRoles',
+	async (
+		{ id, role, token }: { id: string; role: string; token: string },
+		{ rejectWithValue },
```

Comment:
to make code more readable make create separate interface for this params you have passed into asignRoles action